relations/src/r1cs/constraint_system.rs

```diff
@@ -311,12 +311,11 @@ impl<F: Field> ConstraintSystem<F> {
 
         let used_times = self.lc_num_times_used(true);
 
-        // step 2: Modify the lcs accordingly
-        let mut outlined_lcs = BTreeMap::new();
-
-        let mut additional_a_constraints = Vec::<LcIndex>::new();
-        let mut additional_c_constraints_witness_assignments = Vec::<usize>::new();
+        let mut new_witness_linear_combinations = Vec::new();
+        let mut new_witness_indices = Vec::new();
 
+        // `outlined_lcs` stores the modified linear combinations.
+        let mut outlined_lcs = BTreeMap::new();
         for (&index, lc) in self.lc_map.iter() {
             let mut outlined_lc = LinearCombination::new();
             for &(coeff, var) in lc.iter() {
```

Comment:
Can we have a comment to describe what this for loop is doing? It looks `outlined_lcs` is acting as a map to store all prior replacements, and inlining all symbols.

Though I don't get why the core loop works though. (e.g. why does outlined_lcs always contain lc_index in the line `let lc = outlined_lcs.get(&lc_index).expect("should be inlined");`)

Reply:
Resolved by adding more explanations.

Reply:
Thanks! The algorithm now makes sense to me =)